protos/memcache/memcache.go

```diff
@@ -0,0 +1,506 @@
+package memcache
+
+import (
+	"encoding/json"
+	"math"
+	"time"
+
+	"github.com/elastic/libbeat/common"
+	"github.com/elastic/libbeat/logp"
+
+	"github.com/elastic/packetbeat/config"
+	"github.com/elastic/packetbeat/protos/applayer"
+)
+
+// memcache types
+type Memcache struct {
+	Ports   PortsConfig
+	results chan common.MapStr
+	config  parserConfig
+
+	udpMemcache
+	tcpMemcache
+
+	handler MemcacheHandler
+}
+
+type MemcacheHandler interface {
+	onTransaction(t *transaction)
+}
+
+// message actively parsed
+type message struct {
+	// shared
+	applayer.Message
+	next       *message
+	isComplete bool
+
+	command   *commandType
+	isBinary  bool
+	errorMsg  memcacheString
+	casUnique uint64
+	isCas     bool
+
+	// text part
+	commandLine memcacheString
+	rawCommand  []byte
+	rawArgs     []byte
+	noreply     bool
+
+	// binary part
+	opcode  memcacheOpcode
+	status  uint16
+	vbucket uint16
+	opaque  uint32
+	isQuiet bool
+
+	// values
+	keys         []memcacheString
+	flags        uint32
+	exptime      uint32
+	value        uint64
+	value2       uint64
+	ivalue       int64
+	ivalue2      int64
+	str          memcacheString
+	data         memcacheData
+	bytes        uint
+	bytesLost    uint
+	values       []memcacheData
+	count_values uint32
+
+	stats []memcacheStat
+}
+
+type transaction struct {
+	applayer.Transaction
+
+	command *commandType
+
+	request  *message
+	response *message
+}
+
+type memcacheString struct {
+	raw []byte
+}
+
+type memcacheData struct {
+	data []byte
+}
+
+type memcacheStat struct {
+	Name  memcacheString `json:"name"`
+	Value memcacheString `json:"value"`
+}
+
+// Ports configuration
+// TODO: move to libbeat?
+type PortsConfig struct {
+	Ports []int
+}
+
+// debug logging support.
+
+var debug = makeDebug("memcache")
+
+func makeDebug(module string) func(string, ...interface{}) {
```

Comment:
Is this the same as MakeDebug in logp? If yes, I suggest to use the libbeat one.